src/app/destinyTrackerApi/itemListBuilder.ts

```diff
@@ -1,28 +1,30 @@
 import { flatMap } from '../util';
-import _ from 'underscore';
+import * as _ from 'underscore';
 import { ItemTransformer } from './itemTransformer';
+import { ReviewDataCache } from './reviewDataCache';
+import { D1ItemFetchRequest } from '../item-review/destiny-tracker.service';
+import { DimItem } from '../inventory/store/d2-item-factory.service';
 
 /**
  * Translates collections of DIM items into a collection of data almost ready to ship to the DTR API.
  * Generally tailored to work with weapon data.
- *
- * @class ItemListBuilder
  */
-class ItemListBuilder {
+export class ItemListBuilder {
+  _itemTransformer: ItemTransformer;
   constructor() {
     this._itemTransformer = new ItemTransformer();
   }
 
-  _getNewItems(allItems, reviewDataCache) {
+  _getNewItems(allItems: DimItem[], reviewDataCache: ReviewDataCache): D1ItemFetchRequest[] {
     const allDtrItems = allItems.map((item) => this._itemTransformer.translateToDtrWeapon(item));
     const allKnownDtrItems = reviewDataCache.getItemStores();
 
-    const unmatched = _.reject(allDtrItems, (dtrItem) => _.any(allKnownDtrItems, { referenceId: String(dtrItem.referenceId), roll: dtrItem.roll }));
+    const unmatched = _.reject(allDtrItems, (dtrItem) => _.any(allKnownDtrItems, { referenceId: dtrItem.referenceId.toString(), roll: dtrItem.roll }));
 
     return unmatched;
   }
 
-  _getAllItems(stores) {
+  _getAllItems(stores: any[]) {
```

Comment:
`DimStore[]`?

Reply:
If memory serves, this services D1 vendors as well as D1 stores, so not exactly.

I don't super remember where the two diverge to run through different paths.